precode.go

```diff
@@ -57,6 +87,19 @@ func main() {
 	// 4. Собираем числа из каналов outs
 	// ...
 
+	for i := 0; i < NumOut; i++ {
+		wg.Add(1)
+		go func(in <-chan int64, i int) {
```

Comment:
Лучше внутри функции использовать другие имена
```suggestion
		go func(in <-chan int64, j int) {
```